pkgs/development/libraries/rocksdb/default.nix

```diff
@@ -1,86 +1,41 @@
-{ stdenv
-, fetchFromGitHub
-, fixDarwinDylibNames
-, which, perl
+{ stdenv, fetchFromGitHub, lib, cmake, lz4, zlib, zstd, bzip2, enableLite ? false }:
 
-# Optional Arguments
-, snappy ? null, google-gflags ? null, zlib ? null, bzip2 ? null, lz4 ? null
-
-# Malloc implementation
-, jemalloc ? null, gperftools ? null
-
-, enableLite ? false
-}:
-
-let
-  malloc = if jemalloc != null then jemalloc else gperftools;
-  tools = [ "sst_dump" "ldb" "rocksdb_dump" "rocksdb_undump" "blob_dump" ];
-in
 stdenv.mkDerivation rec {
   name = "rocksdb-${version}";
-  version = "5.11.3";
-
-  outputs = [ "dev" "out" "static" "bin" ];
+  version = "5.18.3";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "rocksdb";
     rev = "v${version}";
-    sha256 = "15x2r7aib1xinwcchl32wghs8g96k4q5xgv6z97mxgp35475x01p";
+    sha256 = "1v2slmmr1dsgf8z0qcfg1y9x1al96859rg48b66p9nsawczd5zv9";
   };
 
-  nativeBuildInputs = [ which perl ];
-  buildInputs = [ snappy google-gflags zlib bzip2 lz4 malloc fixDarwinDylibNames ];
-
-  postPatch = ''
-    # Hack to fix typos
-    sed -i 's,#inlcude,#include,g' build_tools/build_detect_platform
-  '';
-
-  # Environment vars used for building certain configurations
-  PORTABLE = "1";
-  USE_SSE = "1";
-  CMAKE_CXX_FLAGS = "-std=gnu++11";
-  JEMALLOC_LIB = stdenv.lib.optionalString (malloc == jemalloc) "-ljemalloc";
-
-  LIBNAME = "librocksdb${stdenv.lib.optionalString enableLite "_lite"}";
-  ${if enableLite then "CXXFLAGS" else null} = "-DROCKSDB_LITE=1";
-
-  buildAndInstallFlags = [
-    "USE_RTTI=1"
-    "DEBUG_LEVEL=0"
-    "DISABLE_WARNING_AS_ERROR=1"
-  ];
-
-  buildFlags = buildAndInstallFlags ++ [
-    "shared_lib"
-    "static_lib"
-  ] ++ tools ;
-
-  installFlags = buildAndInstallFlags ++ [
-    "INSTALL_PATH=\${out}"
-    "install-shared"
-    "install-static"
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ lz4 zlib zstd bzip2 ];
+  checkInputs = [ ];
```

Comment:
Is `checkInputs` needed? Does cmake automatically do some kind of checking? If we need to turn of checking, can we use `doCheck = false;` instead?